### PR TITLE
[RAPTOR-4634] implement DR predictions response, release 1.4.14

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
CM PPS  must return predictions in the same format as DR prediction server. Thus this must be implemented in DRUM.

Add support for `--deployment-config` parameter which expects to receive `deployment_document.json` from model package.
If parameter is provided, DRUM returns prediction response in the DR format.

I tried to reuse generation methods (regression/anomaly/binary/multiclass) from here https://github.com/datarobot/DataRobot/blob/ff431de4b80b873b0e2c1c478452cbeaa8e2d90c/common/prediction/prediction_io.py#L100-L126

### CODE REVIEW

File: custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import json
+
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.common import TargetType
+
+
+def parse_validate_deployment_config_file(filename):
+    if filename is None:
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py:8]: Maybe better `if not filename`? (to include an empty string)

File: custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        return None
+    with open(filename) as f:
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py:10]: What if the `filename` does not exists? Maybe add `try ... except`

File: tests/testdata/deployment_config/anomaly.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "target": {
+        "class_mapping": null, 
+        "missing_maps_to": null, 
+        "name": null, 
+        "prediction_threshold": 0.5, 
+        "type": "Anomaly"
+    }
+}
```
Comment (zohar-mizrahi) [inline tests/testdata/deployment_config/anomaly.json:9]: nit: no newline at end of file

File: tests/testdata/deployment_config/regression.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "target": {
+        "class_mapping": null, 
+        "missing_maps_to": null, 
+        "name": "MEDV", 
+        "prediction_threshold": 0.5, 
+        "type": "Regression"
+    }
+}
```
Comment (zohar-mizrahi) [inline tests/testdata/deployment_config/regression.json:9]: nit: no newline at end of file

File: custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py (hunk 2 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py:10]: filename existence is checked in the arg parse validator

File: custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py (hunk 1 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/resource/deployment_config_helpers.py:8]: filename existence is checked in the arg parse validator

File: tests/testdata/deployment_config/regression.json (hunk 4 above)
Comment (yakov-g) [reply tests/testdata/deployment_config/regression.json]: Done

File: tests/testdata/deployment_config/anomaly.json (hunk 3 above)
Comment (yakov-g) [reply tests/testdata/deployment_config/anomaly.json]: Done

**Review by zohar-mizrahi — APPROVED** [review]

**Review by ikalnytskyi — APPROVED** [review]

**Pull request merged**